gobblin-temporal/src/main/java/org/apache/gobblin/temporal/yarn/GobblinTemporalApplicationMaster.java

```diff
@@ -79,13 +79,14 @@ public GobblinTemporalApplicationMaster(String applicationName, String applicati
             ConfigValueFactory.fromAnyRef(YarnHelixUtils.getContainerNum(containerId.toString()))),
         Optional.<Path>absent());
 
-    String containerLogDir = config.getString(GobblinYarnConfigurationKeys.LOGS_SINK_ROOT_DIR_KEY);
-    GobblinYarnLogSource gobblinYarnLogSource = new GobblinYarnLogSource();
-    if (gobblinYarnLogSource.isLogSourcePresent()) {
-      Path appWorkDir = PathUtils.combinePaths(containerLogDir, GobblinClusterUtils.getAppWorkDirPath(this.clusterName, this.applicationId), "AppMaster");
-      logCopier = gobblinYarnLogSource.buildLogCopier(this.config, containerId.toString(), this.fs, appWorkDir);
-      this.applicationLauncher
-          .addService(logCopier);
+    if (config.hasPath(GobblinYarnConfigurationKeys.LOGS_SINK_ROOT_DIR_KEY)) {
+      String containerLogDir = config.getString(GobblinYarnConfigurationKeys.LOGS_SINK_ROOT_DIR_KEY);
+      GobblinYarnLogSource gobblinYarnLogSource = new GobblinYarnLogSource();
+      if (gobblinYarnLogSource.isLogSourcePresent()) {
+        Path appWorkDir = PathUtils.combinePaths(containerLogDir, GobblinClusterUtils.getAppWorkDirPath(this.clusterName, this.applicationId), "AppMaster");
```

Comment:
do we need anything similar for the TaskRunner?

Reply:
The task runner in temporal has the check for this property already built in